lib/docs/main.js

```diff
@@ -427,15 +427,12 @@
           const member_category = wasm_exports.categorize_decl(member, 0);
           switch (member_category) {
             case CAT_namespace:
-              if (wasm_exports.decl_field_count(member) > 0) {
+              if (!wasm_exports.decl_is_struct(member) || wasm_exports.decl_field_count(member) > 0) {
```

Comment:
seems like this logic should go into `categorize_decl` rather than the UI code, no?

Reply:
That does seem like a cleaner approach. I updated the PR to create a new category, `container`, to represent the "type" case here. Since this distinction is now done entirely in the classification code on the Zig side, I removed the (now unused) `decl_field_count` and its associated state.